File: BSBackEnd/Models/Usuario.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace BSBackEnd.Models
{
    public class Usuario
    {
        public Guid Id { get; set; }

        [Required(ErrorMessage = "Nome obrigatório"), MaxLength(100)]
        public string Nome { get; set; }

        [Required(ErrorMessage = "Email obrigatório"), MaxLength(30), ]
        public string Email { get; set; }

        [Required(ErrorMessage = "Senha obrigatória"), MaxLength(30)]
        public string Senha { get; set; }
        [MaxLength(14)]
        public string CpfCnpj { get; set; }        
        [MaxLength(100)]
        public string Endereco { get; set; }
        [MaxLength(6)]
        public string Numero { get; set; }
        [MaxLength(8)]
        public string Cep { get; set; }
        [MaxLength(50)]
        public string Bairro { get; set; }
        [MaxLength(14)]
        public string Telefone { get; set; }

        public List<Pedido> ComprasRealizadas { get; set; }

        public List<Anuncio> AnunciosRealizados { get; set; }

    }
}

File: bsmobile/lib/models/Usuario.dart
class Usuario{
  String id;
  String nome;
  String email;
  String senha;
  String cpfCnpj;
  String endereco;
  String numero;
  String cep;
  String bairro;
  String telefone;

  Usuario(this.id, this.nome, this.email, this.senha);

  Usuario.fromJson(Map<String, dynamic> json) {
    this.id = json['id'];
    this.nome = json['nome'];
    this.email = json['email'];
    this.senha = json['senha'];
    this.cpfCnpj = json['cpfCnpj'];
    this.endereco = json['endereco'];
    this.numero = json['numero'];
    this.cep = json['cep'];
    this.bairro = json['bairro'];
    this.telefone = json['telefone'];
  }
}